### PR TITLE
fix(qwen3): align bf16 greedy numerics

### PR DESCRIPTION
**Pull request by NolanHo**
## Summary

Fixes Qwen3-4B base greedy drift against HF GPU bf16 by matching the materialization boundaries that affect tie-sensitive logits.

This was discovered while working on Qwen3 LoRA support in #173, where base-model numerical drift had to be separated from LoRA-specific correctness issues. The root-cause issue is tracked in #174.

Changes:

- preserve bf16 residual-add materialization before RMSNorm in Qwen3 prefill/decode/unified paths
- use lower-token-id greedy tie-break for exact logit ties
- split Qwen3 MLP gate/up projections into separate row-sliced GEMMs
- match HF/PyTorch RoPE bf16 elementwise multiply materialization in fused q/k norm+RoPE

## Validation

Local:

```bash
cargo fmt --all --check
PEGAINFER_CUDA_SM=80 cargo check --release -p pegainfer-qwen3-4b
```

GPU worker evidence from the investigation:

- `Tell me a story`, Qwen3-4B, HF CUDA bf16 vs PegaInfer
- no first diff through 128 generated tokens
- no first diff through 256 generated tokens

## Notes

The existing Qwen3 e2e golden text appears to encode the old drift branch, so a follow-up should add or refresh a regression gate for this HF bf16 greedy trace.

### CODE REVIEW

**Review by xiaguan — APPROVED** [review]
LGTM
